dev/tools/localization/gen_l10n.dart

```diff
@@ -290,9 +324,59 @@ String generateDateFormattingLogic(Map<String, dynamic> bundle, String key) {
   return result;
 }
 
-List<String> genIntlMethodArgs(Map<String, dynamic> bundle, String key) {
-  final List<String> attributes = <String>['name: \'$key\''];
-  final Map<String, dynamic> attributesMap = bundle['@$key'] as Map<String, dynamic>;
+String generateNumberFormattingLogic(Map<String, dynamic> arbBundle, String resourceId) {
+  String result = '';
+  final Map<String, dynamic> attributesMap = arbBundle['@$resourceId'] as Map<String, dynamic>;
+  if (attributesMap != null && attributesMap.containsKey('placeholders')) {
+    final Map<String, dynamic> placeholders = attributesMap['placeholders'] as Map<String, dynamic>;
+    String optionalParametersString = '';
+    for (String placeholder in placeholders.keys) {
+      final dynamic value = placeholders[placeholder];
+      if (
+        value is Map<String, dynamic> &&
+        _isProperNumberFormat(value, placeholder)
+      ) {
+        if (value.containsKey('optionalParameters')) {
+          final Map<String, dynamic> optionalParameters = value['optionalParameters'] as Map<String, dynamic>;
+          for (String parameter in optionalParameters.keys)
+            optionalParametersString += '\n      $parameter: ${optionalParameters[parameter]},';
+        }
+
+        result += '''
+
+    final NumberFormat ${placeholder}NumberFormat = NumberFormat.${value['format']}(
+      locale: _localeName,@optionalParameters
+    );
+    final String ${placeholder}String = ${placeholder}NumberFormat.format($placeholder);
+''';
+      }
+    }
+
+    result = result.replaceAll('@optionalParameters', optionalParametersString);
+  }
+
+  return result;
+}
+
+bool _isProperDateFormat(Map<String, dynamic> value, String placeholder) {
+  return _isDateParameter(value)
+      && _containsFormatKey(value, placeholder)
+      && _isDateParameterValid(value, placeholder);
+}
+
+bool _isProperNumberFormat(Map<String, dynamic> value, String placeholder) {
+  return _isNumberParameter(value)
+      && _containsFormatKey(value, placeholder)
+      && _isNumberParameterValid(value, placeholder);
+}
+
+bool _isFormattedPlaceholder(Map<String, dynamic> value, String placeholder) {
```

Comment:
More unnecessary namesmithing: this could be _isVlaidPlaceholder (or _isProper..)